### DISCUSSION TITLE
Accessing repository topics in github pages theme

### DISCUSSION THREAD

**Discussion by DavyCats** [community/community#23326] [Other Feature Feedback, Questions, & Ideas]
<p>Is it possible to retrieve the topics for a repository in a theme?</p>
<p>I would like to list all repositories which have a certain topic, but I haven’t been able to find a way to this. Is anyone aware of a method of doing this? Otherwise it might be a nice feature to add.</p>
<p>An example of what it might look like:</p>
<pre><code>{% for repository in site.github.public_repositories %}
{% if repository.topics contains "production_ready" %}
      &lt;a href="{{ repository.name }}"&gt;{{ repository.name }}&lt;/a&gt; (&lt;a href="{{ repository.html_url }}"&gt;repo&lt;/a&gt;) &lt;/br&gt;
   {% endif %}
{% endfor %}
</code></pre>

**Comment by lee-dohm** [comment]
<p>Something like this would most likely need to be a Jekyll plugin. You could search the list of GitHub repositories that have the <a href="https://github.com/search?q=topic%3Ajekyll-plugin" rel="nofollow noopener">jekyll-plugin topic</a> perhaps? I didn’t find anything pre-made that seemed to fit what you’re going after, but perhaps there is something pre-made that will get you started?</p>
<p>I hope that helps!</p>

**Comment by DavyCats** [comment, marked as answer]
<p>I found an alternative method using the repository description rather than topics, but thanks for the suggestion!</p>
<p>Here is what I’m doing in case anyone is interested :</p>
<p>I’m adding “Category:&lt;label&gt;” at the end of the description and then using this to create arrays of repo names:</p>
<pre><code>{% assign productionReady = '' | split: '' %}
{% assign experimental = '' | split: '' %}

{% for repository in site.github.public_repositories %}
  {% assign category = repository.description | split: "Category:" | last %}
  {% assign repoName = repository.name | split: '@@@' %}
    {% if category == "Production-Ready" %}
      {% assign productionReady = productionReady | concat: repoName %}
    {% endif %}
    {% if category == "Experimental" %}
      {% assign experimental = experimental | concat: repoName %}
    {% endif %}
{% endfor %}
</code></pre>
<p>Then looping over these arrays I can make links.</p>

**Comment by asbjornu** [comment]
<p>I’d love to have topics available in the <a href="https://help.github.com/articles/repository-metadata-on-github-pages/" rel="nofollow noopener">Repository Metadata</a>. I assume the reason it’s not in there is because of <a href="https://developer.github.com/v3/repos/" rel="nofollow noopener">the following disclaimer in the API documentation</a>:</p>
<blockquote>
<p><strong>Note:</strong> The topics property for repositories on GitHub is currently available for developers to preview. To view the topics property in calls that return repository results, you must provide a custom <a href="https://developer.github.com/v3/media" rel="nofollow noopener">media type</a> in the Accept header:</p>
</blockquote>
<blockquote>
<p>application/vnd.github.mercy-preview+json</p>
</blockquote>
<p>It would be neat if we could opt-in to preview features like this with something like preview: true in _config.yml.</p>

**Comment by lee-dohm** [comment]
<p>If you have a feature that you’d like to suggest for <a href="https://jekyllrb.com" rel="nofollow noopener">Jekyll</a>, which is how you would get a change to how the <code>_config.yml</code> behaves, you can file a feature request on the <a href="https://github.com/jekyll/jekyll/issues" rel="nofollow noopener">Jekyll project on GitHub</a>.</p>
<p>I hope that helps!</p>

**Comment by asbjornu** [comment]
<p>Since this feature is only relevant for GitHub Pages, I don’t think Jekyll is the correct recipient of such a feature request. A github node in _config.yml could probably be used to contain GitHub Pages related configuration.</p>

**Comment by asbjornu** [comment]
<p>repository.topics was made available in <a href="https://github.com/jekyll/github-metadata" rel="nofollow noopener">github-metadata</a> version 2.13.0 by the <a href="https://github.com/jekyll/github-metadata/pull/166" rel="nofollow noopener">following pull request</a>.</p>
